test/realm_test.dart

```diff
@@ -463,21 +463,46 @@ Future<void> main([List<String>? args]) async {
       expect(cars, isNotNull);
     });
 
-    test('Results.all() length', () {
+    test('Results length', () {
       var config = Configuration([Car.schema]);
       var realm = Realm(config);
 
       var cars = realm.all<Car>();
       expect(cars.length, 0);
 
-      final car = Car();
-      realm.write(() => realm.add(car));
+      final carOne = Car()..make = "Toyota 1";
+      final carTwo = Car()..make = "Toyota 2";
+      final carThree = Car()..make = "Renault";
+      realm.write(() => realm.addAll([carOne, carTwo, carThree]));
+
+      expect(cars.length, 3);
+
+      final filteredCars = realm.query<Car>('make BEGINSWITH "Toyota"');
+      expect(filteredCars.length, 2);
+
+      realm.write(() => realm.deleteMany(filteredCars));
+      expect(filteredCars.length, 0);
 
       expect(cars.length, 1);
+    });
 
-      realm.write(() => realm.delete(car));
+    test('Results length with query', () {
+      var config = Configuration([Car.schema]);
+      var realm = Realm(config);
 
+      var cars = realm.all<Car>();
       expect(cars.length, 0);
+
+      final carOne = Car()..make = "Toyota";
+      final carTwo = Car()..make = "Toyota";
+      realm.write(() => realm.addAll([carOne, carTwo]));
+
+      expect(cars.length, 2);
+
+      final filteredCars = realm.query<Car>('make == "Toyota"');
+      // TODO: To be discussed. The length here is returned 1, should be 2
```

Comment:
Make is primary key on car, so it is really one object

Reply:
Oh, you are right. But then why it allows me to add two objects with the same key? And also in results coming from the query it returns 2 items, only the length is 1 since it is taken with `realmCore.getResultsCount(this)`

Reply:
Fixed method `add` in realm to throw exception for duplicate primary keys. Commit https://github.com/realm/realm-dart/pull/175/commits/e3083d94da8f9328738f03fcdef7c0daaa555871

Reply:
C API method `realm_object_create_with_primary_key` expects primary key **value** instead of primary key column name.

Reply:
good catch.